src/librustc_yk_sections/emit_tir.rs

```diff
@@ -208,88 +208,66 @@ fn do_generate_tir<'a, 'tcx, 'gcx>(
-        Self {
-            mir,
-            pack,
-            def_sites,
-        }
-    }
-
-    /// Insert PHI nodes, returning the mutated pack.
-    fn pack(mut self) -> ykpack::Pack {
-        let doms = self.mir.dominators();
-        let df = DominatorFrontiers::new(self.mir, &doms);
-        let num_tir_vars = self.def_sites.len();
-
-        // We first need a mapping from block to the variables it defines. Appel calls this
-        // `A_{orig}`. We can derive this from our definition sites.
-        let (a_orig, num_tir_blks) = {
-            let ykpack::Pack::Mir(ykpack::Mir{ref blocks, ..}) = self.pack;
-            let num_tir_blks = blocks.len();
-
-            let mut a_orig: IndexVec<BasicBlock, BitSet<TirLocal>> =
-                IndexVec::from_elem_n(BitSet::new_empty(num_tir_vars), num_tir_blks);
-            for (a, def_blks) in self.def_sites.iter().enumerate() {
-                for bb in def_blks.iter() {
-                    // `def_sites` is guaranteed to have at most `u32::max_value()` items.
-                    a_orig[bb].insert(a as u32);
-                }
-            }
-            (a_orig, num_tir_blks)
-        };
-
-        let mut a_phi: Vec<BitSet<TirLocal>> = Vec::with_capacity(num_tir_blks);
-        a_phi.resize(num_tir_blks, BitSet::new_empty(num_tir_vars));
-        // We don't need the elements of `def_sites` again past this point, so we can take them out
-        // of `def_sites` with a draining iterator and mutate in-place.
-        for (a, mut w) in self.def_sites.drain(..).enumerate() {
-            while !w.is_empty() {
-                let n = bitset_pop(&mut w);
-                for y in df.frontier(n).iter() {
-                    let y_usize = y.index();
-                    // `self.def_sites` is guaranteed to only contain indices expressible by `u32`.
-                    let a_u32 = a as u32;
-                    if !a_phi[y_usize].contains(a_u32) {
-                        // Appel would insert the Phi node here. We use a second pass to keep the
-                        // borrow checker happy (self is already borrowed in this loop).
-                        a_phi[y_usize].insert(a_u32);
-                        if !a_orig[y].contains(a_u32) {
-                            w.insert(y);
-                        }
-                    }
-                }
+/// Algorithm reference:
+/// Bottom of p406 of 'Modern Compiler Implementation in Java (2nd ed.)' by Andrew Appel.
+fn insert_phis(blocks: &mut Vec<ykpack::BasicBlock>, mir: &Mir,
+               mut def_sites: Vec<BitSet<BasicBlock>>)
+{
+    let doms = mir.dominators();
+    let df = DominatorFrontiers::new(mir, &doms);
+    let num_tir_vars = def_sites.len();
+
+    // We first need a mapping from block to the variables it defines. Appel calls this
+    // `A_{orig}`. We can derive this from our definition sites.
+    let (a_orig, num_tir_blks) = {
+        let num_tir_blks = blocks.len();
```

Comment:
This variable name is as long as the RHS expression. I'd personally remove the variable and just use the RHS expression where needed.

Reply:
That code is removed in a later commit.

Reply:
OK.